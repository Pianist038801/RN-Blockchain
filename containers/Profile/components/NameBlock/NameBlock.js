import React from 'react'
import PropTypes from 'prop-types'
import { View, Text } from 'react-native'
import styles from '../../../../style.js'

const NameBlock = (props) => {
  const { name, line2 } = props

  let line2Value = line2
  if (line2 === null || line2 === 'null') {
    line2Value = ''
  } else if (line2.indexOf(' - null') !== -1) {
    line2Value = line2.split(' - null')[0]
  }

  return (
    <View style={styles.profileEntityContent}>
      <Text style={styles.profileEntityPrimaryText}>{name}</Text>
      {line2Value !== '' &&
        <Text>{line2Value}</Text>
      }
    </View>
  )
}

NameBlock.propTypes = {
  name: PropTypes.string.isRequired,
  line2: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.object
  ])
}

export default NameBlock
