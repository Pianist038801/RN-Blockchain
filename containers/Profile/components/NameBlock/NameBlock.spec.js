import React from 'react'
import renderer from 'react-test-renderer'
import NameBlock from './index'

describe('containers/Profile/components/NameBlock', () => {
  describe('when props are good', () => {
    it('renders correctly', () => {
      const props = {
        name: 'hello',
        line2: 'lorem ipsum'
      }

      const tree = renderer.create(
        <NameBlock {...props} />
      ).toJSON()

      expect(tree).toMatchSnapshot()
    })
  })

  describe('when props.line2 is null', () => {
    it('renders correctly', () => {
      const props = {
        name: 'hello',
        line2: null
      }

      const tree = renderer.create(
        <NameBlock {...props} />
      ).toJSON()

      expect(tree).toMatchSnapshot()
    })
  })

  describe('when props.line2 is "null"', () => {
    it('renders correctly', () => {
      const props = {
        name: 'hello',
        line2: 'null'
      }

      const tree = renderer.create(
        <NameBlock {...props} />
      ).toJSON()

      expect(tree).toMatchSnapshot()
    })
  })

  describe('when props.line2 contains "null"', () => {
    it('renders correctly', () => {
      const props = {
        name: 'hello',
        line2: 'Applied Blockchain - null'
      }

      const tree = renderer.create(
        <NameBlock {...props} />
      ).toJSON()

      expect(tree).toMatchSnapshot()
    })
  })
})
