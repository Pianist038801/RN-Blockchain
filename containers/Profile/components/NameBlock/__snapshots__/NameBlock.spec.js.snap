// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`containers/Profile/components/NameBlock when props are good renders correctly 1`] = `
<View
  style={
    Object {
      "marginLeft": 22.5,
      "width": 360,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#569cee",
        "fontWeight": "bold",
      }
    }
  >
    hello
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    lorem ipsum
  </Text>
</View>
`;

exports[`containers/Profile/components/NameBlock when props.line2 contains "null" renders correctly 1`] = `
<View
  style={
    Object {
      "marginLeft": 22.5,
      "width": 360,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#569cee",
        "fontWeight": "bold",
      }
    }
  >
    hello
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    Applied Blockchain
  </Text>
</View>
`;

exports[`containers/Profile/components/NameBlock when props.line2 is "null" renders correctly 1`] = `
<View
  style={
    Object {
      "marginLeft": 22.5,
      "width": 360,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#569cee",
        "fontWeight": "bold",
      }
    }
  >
    hello
  </Text>
</View>
`;

exports[`containers/Profile/components/NameBlock when props.line2 is null renders correctly 1`] = `
<View
  style={
    Object {
      "marginLeft": 22.5,
      "width": 360,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#569cee",
        "fontWeight": "bold",
      }
    }
  >
    hello
  </Text>
</View>
`;
